src/module_1/module_1_meteo_api.py

```
@@ -1,7 +1,101 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import matplotlib.pyplot as plt
+import time
 
+
+#Base URL for Open Meteo API
+API_URL = "https://archive-api.open-meteo.com/v1/archive"
+#Coordinates of the cities
+COORDINATES = {
+    "Madrid": {"latitude": 40.42, "longitude": -3.70},
+    "London": {"latitude": 51.51, "longitude": -0.13},
+     "Rio": {"latitude": -22.91, "longitude": -43.17},
+}
+
+#Varibles to extract
+VARIABLES = ["temperature_2m_mean", "precipitation_sum", "wind_speed_10m_max"]
+
+#Function to call the API
+def get_data_meteo_API(city, start_year, end_year):
+   #Variables latitude and longitude to be able to then recover multiple cities with a for
+   latitude = COORDINATES[city]["latitude"]
+   longitude = COORDINATES[city]["longitude"]
+
+   #Array to stock all the data that will be recover from the API
+   all_data = []
+
+   #For to pass through all the years in the selected interval
+   for year in range(start_year, end_year):
+      params = {
+         "latitude" : latitude,
+         "longitude" : longitude,
+         "start_date" : f"{year}-01-01",
+         "end_date" : f"{year}-12-31",
+         "daily" : VARIABLES,
+         "timezone" : "Europe/Madrid"
+
+      }
+
+      #Request to the API all the parameters wished
+      response = requests.get(API_URL, params=params)
+
+      if response.status_code == 200:
+         data = response.json()
+         all_data.append (pd.DataFrame(data["daily"],columns = ["time"]+ VARIABLES))
+        
+      
+      else:
+         print(f"Error fetching data for {city} in {year} also {start_year} and {end_year}. Status Code: {response.status_code}")
```

I think you should improve the exception management here. If one of your API requests fail, what should the program do? In this case, it will continue fetching data and will plot an incomplete graph. Personally, I prefer to raise an exception to make it fail fast and try to solve the problem.